includes/cli/class-data-backfill.php

```diff
@@ -0,0 +1,365 @@
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		add_action( 'init', [ __CLASS__, 'register_commands' ] );
+	}
+
+	/**
+	 * Register the WP-CLI commands
+	 *
+	 * @return void
+	 */
+	public static function register_commands() {
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			WP_CLI::add_command( 'newspack-network data-backfill', [ __CLASS__, 'data_backfill' ] );
+		}
+	}
+
+	/**
+	 * Find existing webhook requests for a given action and data.
+	 *
+	 * @param string $action The action name.
+	 * @param int    $timestamp The timestamp.
+	 * @param array  $data The data.
+	 */
+	private static function find_webhook_requests( $action, $timestamp, $data ) {
+		return get_posts(
+			[
+				'post_type'   => \Newspack\Data_Events\Webhooks::REQUEST_POST_TYPE,
+				'post_title'  => $action,
+				'post_status' => 'any',
+				'meta_query'  => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+					'relation' => 'AND',
+					[
+						'key'     => 'timestamp',
+						'value'   => $timestamp,
+						'compare' => '=',
+					],
+					[
+						'key'     => 'action_name',
+						'value'   => $action,
+						'compare' => '=',
+					],
+					[
+						'key'     => 'data',
+						'value'   => wp_json_encode( $data ),
+						'compare' => '=',
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Process reader_registered events.
+	 *
+	 * @param string $start The start date.
+	 * @param string $end The end date.
+	 * @param bool   $live Whether to run in live mode.
+	 * @param bool   $verbose Whether to output verbose information.
+	 */
+	private static function process_reader_registered( $start, $end, $live, $verbose ) {
+		// Get all users registered between $start and $end.
+		$users = get_users(
+			[
+				'role__in'   => \Newspack\Reader_Activation::get_reader_roles(),
+				'date_query' => [
+					'after'     => $start,
+					'before'    => $end,
+					'inclusive' => true,
+				],
+				'fields'     => [ 'id', 'user_email', 'user_registered' ],
+				'number'     => -1,
+				'meta_query' => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+					[
+						'key'     => \Newspack_Network\Utils\Users::USER_META_REMOTE_SITE,
+						'compare' => 'NOT EXISTS',
+					],
+				],
+			]
+		);
+		if ( ! $verbose ) {
+			self::$progress = \WP_CLI\Utils\make_progress_bar( 'Processing users', count( $users ) );
+		}
+
+		foreach ( $users as $user ) {
+			$registration_method = get_user_meta( $user->ID, \Newspack\Reader_Activation::REGISTRATION_METHOD, true );
+			$user_data = [
+				'user_id'  => $user->ID,
+				'email'    => $user->user_email,
+				'metadata' => [
+					// 'current_page_url' is not saved, can't be backfilled.
+					'registration_method' => empty( $registration_method ) ? 'backfill-script' : $registration_method,
+				],
+			];
+			if ( $live ) {
+				self::process_event_entity( $user_data, strtotime( $user->user_registered ), 'reader_registered' );
+			}
+			if ( $verbose ) {
```

Comment:
apparently when you run `live`, the progress bar overrides the verbose output. Maybe only show progress bar if verbose is false

Reply:
`--verbose` and progress bar are mutually exclusive, independent of `--live`. If the progress bar is rendered, logging anything would break the progress bar rendering. I haven't found a way to render the progress bar _and_ at the same time log individual lines.

Reply:
yesterday I thought `--live` was enforcing the progress bar. But today it's working as expected.. my mistake